scripts/server

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+if [[ -n "${ASHLAR_BLUEPRINT_DEBUG}" ]]; then
+    set -x
+fi
+
+function usage() {
+    echo -n "Usage: $(basename "$0")
+
+Starts servers using docker-compose.
+"
+}
+
+if [ "${BASH_SOURCE[0]}" = "${0}" ]
+then
+    if [ "${1:-}" = "--help" ]
+    then
+        usage
+    else
+        if [[ $(sudo netstat -nlp | grep :::8000) ]];
+        then
+            echo "It appears you already have a process running on port 8000:"
+            echo
+            echo $(sudo netstat -nlp | grep :::8000)
+            echo
+            echo "Terminate that process, or alter docker-compose.yml to run the app on another port."
+            exit 1
+        else
+            docker-compose run --service-ports ashlar-server
```

Comment:
I don't think I used `--service-ports` when I tested this out earlier; does it work without it?

Reply:
It doesn't work on my machine without  `--service-ports` :confused: I get a "Connection failed" error in the browser and can't curl/ping the host. I went digging to see if maybe it's a difference between our docker-compose versions, but it seems like this has been the default behavior since even before `--service-ports` was introduced: https://github.com/docker/compose/issues/1256 I wonder if you ran `docker-compose up` instead? Or perhaps you have an alias set up? But if you're able to reproduce it that would be super interesting!

Reply:
Oh, yeah! I was running `docker-compose up` by instinct and I totally missed that this uses `run`, everything makes sense now. Any particular reason you chose `docker-compose run`? We usually use that for one-off commands (like in `scripts/update`) rather than for bringing up the whole application stack.

Reply:
I only used it in this PR because I was having trouble getting the `db` and `ashlar-server` services up and running in the right sequence, but I'm defaulting to `docker-compose up` in the schema editor PR that I'm working on now!